front/lib/api/assistant/actions/visualization.ts

```diff
@@ -175,16 +180,61 @@ export class VisualizationConfigurationServerRunner extends BaseActionConfigurat
       }),
     };
 
-    // Turn the conversation into a digest that can be presented to the model.
+    const contentFragmentsText = await Promise.all(
+      conversation.content
+        .flat(1)
+        .filter((m) => isContentFragmentType(m))
+        .map(async (m) => {
+          // This is needed because the ContentFragmentType is not actually narrowed correctly,
+          // this type is actually pretty complex and even if VSCode figure it out, `npm run build` does not.
+          // @flavien knows about this.
+          assert(isContentFragmentType(m));
```

Comment:
can we remove this ?

Reply:
We should use an `as` or something, it doesn't make sense to have assert here

Reply:
Don't we have this snippet logic somewhere already ?

Reply:
We do have some snippets generated but here, it computes different snippets.